src/tests/test_api.py

```diff
@@ -195,6 +196,13 @@ def validate_namespace_schema(ns):
             assert isinstance(ns["usetokenonread"], bool)
             assert ns["dirlisthost"] is None or PROTOCOL_HOST_PORT_RE.match(ns["dirlisthost"])
             assert ns["writebackhost"] is None or PROTOCOL_HOST_PORT_RE.match(ns["writebackhost"])
+            credgen = ns["credential_generation"]
+            if credgen is not None:
+                assert isinstance(credgen["max_scope_depth"], int) and credgen["max_scope_depth"] > -1
+                assert credgen["strategy"] in CredentialGeneration.STRATEGIES
+                assert credgen["issuer"] and "://" in credgen["issuer"]
```

Comment:
Future idea: you can always parse this as a URL and ensure `netloc` isn't empty and `https` is the scheme:

```
>>> import urllib.parse
>>> urllib.parse.urlparse("https://foo.com")
ParseResult(scheme='https', netloc='foo.com', path='', params='', query='', fragment='')
```

Reply:
Done.